### ISSUE TITLE
Object mapping should be resolved at the API request manager level, not in view controllers

### ISSUE THREAD

**Issue by olivierobert** [anishparajuli555/Survey#5]
To separate concerns, the mapping of JSON response into models should be handled in the class that handle API requests. Currently, it's managed in the view controllers which has the following disadvantages: 

1. If the same API call is required elsewhere, it will require to duplicate the code
2. The transformation library `ObjectMapper` is tightly coupled in every single view making it hard to maintain over time. If we were to change the object mapping library e.g. `Mantle` then it would required to refactor all view controllers. But if it was defined only in the class that managed API requests, it will be handled only in one place. Views should manage objects not API request transformation.

**Comment by anishparajuli555** [comment]
Yes..I used to do the same.I had implemented API caller with generics that used to map json in the API Caller side..But didnot used it as the json had not any fixed structure..
